src/main/scala/Covid19/Sources/WorldSource.scala

```diff
@@ -61,4 +60,57 @@ final class Jhu(implicit backend: SttpBackend[Identity, Nothing, NothingT], impl
       case (name, _) => name
     }.toList
   }
-}
\ No newline at end of file
+
+  override def getInfected: IO[Protocol.Response] = {
+    for {
+      confFib <- getSummaryByCategory[Confirmed].start
+      deadFib <- getSummaryByCategory[Dead].start
+      recFib <- getSummaryByCategory[Recovered].start
+
+      recovered <- recFib.join
+      dead <- deadFib.join
+      confirmed <- confFib.join
+
+      items = fuseSummary(confirmed, dead, recovered)
+    } yield new Protocol.Response(items)
+  }
+
+  def fuseSummary(confirmed: Map[String, Confirmed], dead: Map[String, Dead], recovered: Map[String, Recovered]): List[CovidData] = {
+    val keys = (confirmed.keys ++ dead.keys ++ recovered.keys).toSet
+    keys.map { key =>
+      val location = countries.find{ case (_, code) => code == key }.map(_._1).getOrElse("")
+      val confByKey = confirmed.getOrElse(key, Confirmed(key, 0))
+      val deadByKey = dead.getOrElse(key, Dead(key, 0))
+      val recByKey = recovered.getOrElse(key, Recovered(key, 0))
+
+      CovidData(location, key, confByKey, recByKey, deadByKey)
+    }.toList
+  }
+
+  private def getRequest[C <: InfectedCategory: CategoryName] =
+    IO.pure(basicRequest.get(uri"${baseUrl}time_series_covid19_${CategoryName[C].name}_global.csv"))
```

Comment:
better to have that stuff as val

Reply:
How will I pass type parameter in case of val?